thsvs_simpler_vulkan_synchronization.h

```diff
@@ -1054,10 +1058,6 @@ void thsvsGetVulkanImageMemoryBarrier(
 #endif
             pVkBarrier->oldLayout = layout;
         }
-
-#ifdef THSVS_ERROR_CHECK_COULD_USE_GLOBAL_BARRIER
-    assert(pVkBarrier->srcQueueFamilyIndex != pVkBarrier->dstQueueFamilyIndex);

```

Comment:
Rather than outright removing this, I think it'd make sense to just extend the assert to check the layout as well? I.e.

```c
assert((pVkBarrier->srcQueueFamilyIndex != pVkBarrier->dstQueueFamilyIndex) ||
       (pVkBarrier->oldLayout != pVkBarrier->newLayout));
```

Reply:
Exactly, that is already done at line 1111.

Reply:
Ah so it is - wasn't in the diff and I haven't touched this for a while so forgot. Ok I'll resolve this then - thanks!